File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.py
# Last updated: 24/07/2026, 23:55:06
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        flag=True
        i=0
        j=len(needle)
        while flag:
            if j<=len(haystack):
                if haystack[i:j]==needle:
                    flag=False
                    return i
                else:
                    i+=1
                    j+=1
            else:
                return -1

File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.java
// Last updated: 24/07/2026, 23:55:04
class Solution {
    public int strStr(String haystack, String needle) {
        boolean flag = true;
        int i=0;
        int j=needle.length();
        while(flag){
            if (j<=haystack.length()){
                if(haystack.substring(i,j).equals(needle)){
                    flag=false;
                    return i;
                }
                else{
                    i++;
                    j++;
                }
            }
            else{
                return -1;
            }
        }
        return -1;
    }
}